### PR TITLE
testing/dnsdist: new aport

### PR DESCRIPTION
**Pull request by tcely**
https://dnsdist.org
dnsdist is a highly DNS-, DoS-, and abuse-aware loadbalancer.

This is a reworked version of #2755

Depends on #3965

### CODE REVIEW

File: testing/dnsdist/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		--sysconfdir=/etc \
+		--mandir=/usr/share/man \
+		--enable-unit-tests \
+		--enable-dnscrypt \
+		--enable-libsodium \
+		--enable-re2 \
+		--with-protobuf \
+		--with-boost=/usr/include \
+		--with-net-snmp \
+		--with-luajit
```
Comment (ncopa) [inline testing/dnsdist/APKBUILD]: luajit is not supported on s390x so we need to either have `arch="... !s390x"` or use `--without-luajit` on s390x.

Comment (tcely) [reply testing/dnsdist/APKBUILD]: Is regular `lua` supported on `s390x`?

Comment (tcely) [reply testing/dnsdist/APKBUILD]: Dropped back to `lua5.1-dev` for `s390x` and kept `luajit-dev` for everything else.

**Comment by BernhardGruen** [comment]
It seems this aport includes parts of #3965 . Is this correct?

**Comment by tcely** [comment]
@BernhardGruen `v1.3.0` added features using `libfstrm` gated behind the `--enable-fstrm` configure flag. To allow the Travis build to work correctly, I merged the other branch before pushing the changes in https://github.com/alpinelinux/aports/pull/3321/commits/eaea87f8240cb1e713b9671a286dc18502fb24ca

**Comment by tcely** [comment]
@ncopa Removed fstrm commit and squashed the rest.

**Comment by tcely** [comment]
TravisCI passes now that `fstrm` aport is available.

**Pull request closed without merging**